### PR TITLE
Add deploy_build_before example hook for theme assets

### PR DESCRIPTION
**Pull request by retlehs**
not everyone using trellis is going to have a theme that needs this, but i would still like this to exist as a placeholder.

see https://discourse.roots.io/t/deployment-with-the-new-trellis-hooks/5016/2?u=benword

### CODE REVIEW

**Comment by retlehs** [comment]
another benefit is that this also provides an example of how to use a deploy hook along with where to put it

File: deploy-hooks/build-before.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#   args:
+#     chdir: "{{ project.local_path }}/web/app/themes/sage"
+# - name: Run gulp
+#   command: gulp --production
+#   connection: local
+#   args:
+#     chdir: "{{ project.local_path }}/web/app/themes/sage"
+# - name: Copy project local files
+#   synchronize: src="{{ item.src }}"
+#                dest="{{ deploy_helper.new_release_path }}/{{ item.dest }}"
```
Comment (swalkinshaw) [inline deploy-hooks/build-before.yml]: need to use new syntax here

**Comment by fullyint** [comment]
Could this PR also remove the the existing "[Copy project local files](https://github.com/roots/trellis/blob/6e277a4d4106a3dd66ee19cce48c4f9cef32b92c/roles/deploy/tasks/build.yml#L4-L11)" task and the corresponding [`project_local_files`](https://github.com/roots/trellis/blob/6e277a4d4106a3dd66ee19cce48c4f9cef32b92c/roles/deploy/defaults/main.yml#L28-L37) default variable? They seem obsolete now that this PR's "Copy project local files" task demonstrates the new recommended approach of using deploy hooks.

**Comment by swalkinshaw** [comment]
@fullyint good catch! Definitely need to remove those.

**Pull request merged**